src/httpauth/digest.c

```diff
@@ -410,3 +413,189 @@ int httpauth_digest_response_encode(const struct httpauth_digest_resp *resp,
 	mbuf_set_pos(mb, 0);
 	return err;
 }
+
+
+static void httpauth_digest_chall_req_destructor(void *arg)
+{
+	struct httpauth_digest_chall_req *req = arg;
+
+	mem_deref(req->realm);
+	mem_deref(req->domain);
+	mem_deref(req->nonce);
+	mem_deref(req->opaque);
+	mem_deref(req->algorithm);
+	mem_deref(req->qop);
+	mem_deref(req->charset);
+}
+
+
+static int generate_nonce(char **pnonce, const time_t ts,
+	const char *etag, const char *secret)
+{
+	struct mbuf *mb = NULL;
+	char *nonce = NULL;
+	uint8_t *hash = NULL;
+	uint8_t hashlen = SHA256_DIGEST_LENGTH;
+	int err = 0;
+
+	mb = mbuf_alloc(32);
+	if (!mb)
+		return ENOMEM;
+
+	hash = mem_zalloc(hashlen, NULL);
+	if (!hash) {
+		err = ENOMEM;
+		goto out;
+	}
+
+	if (str_isset(secret))
+		err = mbuf_printf(mb, "%lu:%s:%s", ts, etag, secret);
+	else
+		err = mbuf_printf(mb, "%lu:%s", ts, etag);
```

Comment:
printing `time_t` needs a casting like here: https://github.com/baresip/re/commit/c025aec52c816c37d6fa1e9305a2d57aeef8aa0d

Reply:
Thanks for the info.

I added the casts for the `time_t`.